src/main/java/com/structures/helpers/Structure.java

```diff
@@ -0,0 +1,422 @@
+    {
+        InputStream inputstream = MinecraftServer.class.getResourceAsStream("/assets/" + Constants.MOD_ID + "/schematics/" + structureName + ".nbt");
+
+        if(world == null || world.isRemote)
+        {
+            this.settings = settings;
+            this.mc = Minecraft.getMinecraft();
+        }
+        if (inputstream == null)
+        {
+            return;
+        }
+
+        try
+        {
+            this.template = readTemplateFromStream(inputstream);
+        }
+        catch (IOException e)
+        {
+            Log.getLogger().warn(String.format("Failed to load template %s", structureName), e);
+        }
+        finally
+        {
+            IOUtils.closeQuietly(inputstream);
+        }
+    }
+
+    /**
+     * reads a template from an inputstream
+     */
+    private static Template readTemplateFromStream(InputStream stream) throws IOException
+    {
+        NBTTagCompound nbttagcompound = CompressedStreamTools.readCompressed(stream);
+        Template template = new Template();
+        template.read(nbttagcompound);
+        return template;
+    }
+
+    public boolean doesExist()
+    {
+        return template == null ? false : true;
+    }
+
+    public Template.BlockInfo[] getBlockInfo()
+    {
+        Template.BlockInfo[] blockList = new Template.BlockInfo[template.blocks.size()];
+        blockList = template.blocks.toArray(blockList);
+        return blockList;
+    }
+
+    public Template.BlockInfo[] getBlockInfoWithSettings(PlacementSettings settings)
+    {
+        Template.BlockInfo[] blockList = new Template.BlockInfo[template.blocks.size()];
+        blockList = template.blocks.toArray(blockList);
+
+        for (int i = 0; i < blockList.length; i++)
+        {
+            IBlockState finalState = blockList[i].blockState.withMirror(settings.getMirror()).withRotation(settings.getRotation());
+            BlockPos finalPos = Template.transformedBlockPos(settings, blockList[i].pos);
+            Template.BlockInfo finalInfo = new Template.BlockInfo(finalPos, finalState, blockList[i].tileentityData);
+            blockList[i] = finalInfo;
+        }
+        return blockList;
+    }
+
+    public Entity[] getEntityInfo(World world, BlockPos pos)
+    {
+        Template.EntityInfo[] entityInfoList = new Template.EntityInfo[template.entities.size()];
+        entityInfoList = template.blocks.toArray(entityInfoList);
+
+        Entity[] entityList = null;
+
+        for (int i = 0; i < entityInfoList.length; i++)
+        {
+            Entity finalEntity = EntityList.createEntityFromNBT(entityInfoList[i].entityData, world);
+            Vec3d entityVec = entityInfoList[i].pos.add(new Vec3d(pos));
+            finalEntity.setPosition(entityVec.xCoord, entityVec.yCoord, entityVec.zCoord);
+        }
+
+        return entityList;
+    }
+
+    public Entity[] getEntityInfoWithSettings(World world, BlockPos pos, PlacementSettings settings)
```

Comment:
![MINOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-minor.png) Document this public method. [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](http://home.kk-sc.de:9000/coding_rules#rule_key=squid%3AUndocumentedApi)